# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1304361956 25200
#      Mon May 02 11:45:56 2011 -0700
# Node ID e205ebe2ba67aeef7e62cd068df9c6e3d1632a8c
# Parent  55430618bd5fa34e4020232cefa0627c322d1779
Deprecate sylvester_matrix.m function

* NEWS: Add to list of deprecated functions
* matrix.txi: Remove from list of special matrices
* deprecated/module.mk, special-matrix/module.mk: Move function
to deprecated directory.

diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -27,16 +27,17 @@ Summary of important user-visible change
       spchol2inv          split        unmark_rawcommand
 
     The following functions have been deprecated in Octave 3.6 and will
     be removed from Octave 3.10 (or whatever version is the second major
     release after 3.6):
 
       is_duplicate_entry
       studentize
+      sylvester_matrix
 
 
 Summary of important user-visible changes for version 3.4:
 ---------------------------------------------------------
 
  ** BLAS and LAPACK libraries are now required to build Octave.  The
     subset of the reference BLAS and LAPACK libraries has been removed
     from the Octave sources.
diff --git a/doc/interpreter/matrix.txi b/doc/interpreter/matrix.txi
--- a/doc/interpreter/matrix.txi
+++ b/doc/interpreter/matrix.txi
@@ -249,15 +249,13 @@ The following functions return famous ma
 @DOCSTRING(invhilb)
 
 @DOCSTRING(magic)
 
 @DOCSTRING(pascal)
 
 @DOCSTRING(rosser)
 
-@DOCSTRING(sylvester_matrix)
-
 @DOCSTRING(toeplitz)
 
 @DOCSTRING(vander)
 
 @DOCSTRING(wilkinson)
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -19,16 +19,17 @@ deprecated_FCN_FILES = \
   deprecated/perror.m \
   deprecated/replot.m \
   deprecated/saveimage.m \
   deprecated/setstr.m \
   deprecated/sphcat.m \
   deprecated/spvcat.m \
   deprecated/strerror.m \
   deprecated/studentize.m \
+  deprecated/sylvester_matrix.m \
   deprecated/values.m \
   deprecated/weibcdf.m \
   deprecated/weibinv.m \
   deprecated/weibpdf.m \
   deprecated/weibrnd.m
 
 FCN_FILES += $(deprecated_FCN_FILES)
 
diff --git a/scripts/special-matrix/sylvester_matrix.m b/scripts/deprecated/sylvester_matrix.m
rename from scripts/special-matrix/sylvester_matrix.m
rename to scripts/deprecated/sylvester_matrix.m
--- a/scripts/special-matrix/sylvester_matrix.m
+++ b/scripts/deprecated/sylvester_matrix.m
@@ -28,16 +28,23 @@
 ##
 ## @seealso{toeplitz, hankel}
 ## @end deftypefn
 
 ## Author: jwe
 
 function retval = sylvester_matrix (k)
 
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "sylvester_matrix is obsolete and will be removed from a future version of Octave; please use hadamard(2^k) instead");
+  endif
+
   if (nargin != 1)
     print_usage ();
   endif
 
   if (isscalar (k))
     if (k < 1)
       retval = 1;
     else
diff --git a/scripts/special-matrix/module.mk b/scripts/special-matrix/module.mk
--- a/scripts/special-matrix/module.mk
+++ b/scripts/special-matrix/module.mk
@@ -3,17 +3,16 @@ FCN_FILE_DIRS += special-matrix
 special_matrix_FCN_FILES = \
   special-matrix/hadamard.m \
   special-matrix/hankel.m \
   special-matrix/hilb.m \
   special-matrix/invhilb.m \
   special-matrix/magic.m \
   special-matrix/pascal.m \
   special-matrix/rosser.m \
-  special-matrix/sylvester_matrix.m \
   special-matrix/toeplitz.m \
   special-matrix/vander.m \
   special-matrix/wilkinson.m
 
 FCN_FILES += $(special_matrix_FCN_FILES)
 
 PKG_ADD_FILES += special-matrix/PKG_ADD
 
